source/pervasive/seq_lib.rs

```diff
@@ -31,14 +33,101 @@ impl<A> Seq<A> {
     /// thereby 1 smaller.
     ///
     /// If the input sequence is empty, the result is meaningless and arbitrary.
-
     pub open spec fn drop_last(self) -> Seq<A>
         recommends self.len() >= 1
     {
         self.subrange(0, self.len() as int - 1)
-    } 
+    }
+
+    /// returns `true` if the sequequence has no duplicate elements
+    pub open spec fn no_duplicates(self) -> bool {
+        forall(|i, j| (0 <= i < self.len() && 0 <= j < self.len() && i != j)
+            ==> self[i] != self[j])
+    }
+
+    /// Returns `true` if two sequences are disjoint
+    pub open spec fn disjoint(self, other: Self) -> bool {
+        forall |a: A| self.contains(a) ==> !other.contains(a)
+    }
+
+    /// Converts a sequence into a set
+    pub open spec fn to_set(self) -> Set<A> {
+        Set::new(|a: A| self.contains(a))
+    }
+}
+
+
+/// recursive definition of seq to set conversion
+spec fn seq_to_set_rec<A>(seq: Seq<A>) -> Set<A>
+    decreases seq.len() when seq.len() >= 0
```

Comment:
I don't think you the `when seq.len() >= 0` here because `seq.len() >= 0` is always true.

Reply:
Removed the `when` clause.